File: nodejs_backend_todo-main/models/notification.model.js
// models/notification.model.js
const mongoose = require('mongoose');
const db = require('../config/db');
const { Schema } = mongoose;

const notificationSchema = new Schema({
    title: {
        type: String,
        required: true
    },
    description: {
        type: String,
        required: true
    },
    timestamp: {
        type: Date,
        default: Date.now,
        required: true
    }
}, { timestamps: true });

const NotificationModel = db.model('notification', notificationSchema);

module.exports = { NotificationModel };

File: lib/notifications/Notification_Model.dart
// notification_model.dart
class NotificationModel {
  final String title;
  final String description;
  final DateTime timestamp;

  NotificationModel({required this.title, required this.description, required this.timestamp});

  factory NotificationModel.fromJson(Map<String, dynamic> json) {
    return NotificationModel(
      title: json['title'],
      description: json['description'],
      timestamp: DateTime.parse(json['timestamp']),
    );
  }
}
